src/component/Text.tsx

```diff
@@ -160,132 +151,106 @@ const getWordsWithoutCalculate = (children: React.ReactNode): Array<Words> => {
   return [{ words }];
 };
 
-const getWordsByLines = (props: Props, needCalculate: boolean) => {
+const getWordsByLines = (props: Props) => {
   // Only perform calculations if using features that require them (multiline, scaleToFit)
   if ((props.width || props.scaleToFit) && !Global.isSsr) {
     let wordsWithComputedWidth: Array<WordWithComputedWidth>, spaceWidth: number;
 
-    if (needCalculate) {
-      const wordWidths = calculateWordWidths(props);
+    const wordWidths = calculateWordWidths(props);
 
-      if (wordWidths) {
-        const { wordsWithComputedWidth: wcw, spaceWidth: sw } = wordWidths;
+    if (wordWidths) {
+      const { wordsWithComputedWidth: wcw, spaceWidth: sw } = wordWidths;
 
-        wordsWithComputedWidth = wcw;
-        spaceWidth = sw;
-      } else {
-        return getWordsWithoutCalculate(props.children);
-      }
-
-      return calculateWordsByLines(props, wordsWithComputedWidth, spaceWidth, props.width, props.scaleToFit);
+      wordsWithComputedWidth = wcw;
+      spaceWidth = sw;
+    } else {
+      return getWordsWithoutCalculate(props.children);
     }
+
+    return calculateWordsByLines(props, wordsWithComputedWidth, spaceWidth, props.width, props.scaleToFit);
   }
   return getWordsWithoutCalculate(props.children);
 };
 
-export class Text extends Component<Props, State> {
-  static defaultProps = {
-    x: 0,
-    y: 0,
-    lineHeight: '1em',
-    capHeight: '0.71em', // Magic number from d3
-    scaleToFit: false,
-    textAnchor: 'start',
-    verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
-    fill: '#808080',
-  };
-
-  state: State = {};
-
-  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
-    if (
-      nextProps.width !== prevState.prevWidth ||
-      nextProps.scaleToFit !== prevState.prevScaleToFit ||
-      nextProps.children !== prevState.prevChildren ||
-      nextProps.style !== prevState.prevStyle ||
-      nextProps.breakAll !== prevState.prevBreakAll
-    ) {
-      const needCalculate =
-        nextProps.children !== prevState.prevChildren ||
-        nextProps.style !== prevState.prevStyle ||
-        nextProps.breakAll !== prevState.prevBreakAll;
-
-      return {
-        prevWidth: nextProps.width,
-        prevScaleToFit: nextProps.scaleToFit,
-        prevChildren: nextProps.children,
-        prevStyle: nextProps.style,
-        wordsByLines: getWordsByLines(nextProps, needCalculate),
-      };
-    }
+const textDefaultProps = {
+  x: 0,
+  y: 0,
+  lineHeight: '1em',
+  capHeight: '0.71em', // Magic number from d3
+  scaleToFit: false,
+  textAnchor: 'start',
+  verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
+  fill: '#808080',
+};
 
+export const Text = (props: Props) => {
+  const wordsByLines: Array<Words> = useMemo(() => {
+    return getWordsByLines(props);
+  }, [props.width, props.scaleToFit, props.children, props.style, props.breakAll]);
```

Comment:
can we disable the exhaustive dependencies hook here with a justification if it doesn't apply?

Reply:
Btw for the disabling, I do not recommend it, in my workspace I've seen some nasty hidden bugs related to stale closures, I think its essential to have the rule enabled.

Reply:
Agreed, but at times there are valid use-cases for disabling the rule. Wanted the justification for the warning if it was warranted, the refactoring solves this. Thanks!